sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -891,3 +896,56 @@ case class GpuCollectSet(
       inputs: Seq[(ColumnVector, Int)]): RollingAggregationOnColumn =
     RollingAggregation.collectSet().onColumn(inputs.head._2)
 }
+
+trait CpuToGpuAggregateBufferConverter {
+  def createExpression(child: Expression): CpuToGpuBufferTransition
+}
+
+trait GpuToCpuAggregateBufferConverter {
+  def createExpression(child: Expression): GpuToCpuBufferTransition
+}
+
+trait CpuToGpuBufferTransition extends UnaryExpression with CodegenFallback
+
+trait GpuToCpuBufferTransition extends UnaryExpression with CodegenFallback {
+  override def dataType: DataType = BinaryType
+}
+
+class CpuToGpuCollectBufferConverter(
+    elementType: DataType) extends CpuToGpuAggregateBufferConverter {
+  def createExpression(child: Expression): CpuToGpuBufferTransition = {
+    CpuToGpuCollectBufferTransition(child, elementType)
+  }
+}
+
+case class CpuToGpuCollectBufferTransition(
+    override val child: Expression,
+    private val elementType: DataType) extends CpuToGpuBufferTransition {
+
+  private lazy val row = new UnsafeRow(1)
+
+  override def dataType: DataType = ArrayType(elementType, containsNull = false)
+
+  override protected def nullSafeEval(input: Any): ArrayData = {
+    val bytes = input.asInstanceOf[Array[Byte]]
+    row.pointTo(bytes, bytes.length)
+    row.getArray(0).copy()
+  }
+}
+
+class GpuToCpuCollectBufferConverter extends GpuToCpuAggregateBufferConverter {
+  def createExpression(child: Expression): GpuToCpuBufferTransition = {
+    GpuToCpuCollectBufferTransition(child)
+  }
+}
+
+case class GpuToCpuCollectBufferTransition(
+    override val child: Expression) extends GpuToCpuBufferTransition {
+
+  private lazy val projection = UnsafeProjection.create(Array(child.dataType))
+
+  override protected def nullSafeEval(input: Any): Array[Byte] = {
```

Comment:
Again some comments might be nice to point to the Spark code that is doing the serialization/deserialization so we can show how we match.